hazelcast/include/hazelcast/client/config/ClientNetworkConfig.h

```diff
@@ -89,11 +89,112 @@ namespace hazelcast {
+                 */
+                ClientNetworkConfig &setConnectionAttemptLimit(int32_t connectionAttemptLimit);
+
+                /**
+                 * Period for the next attempt to find a member to connect.
+                 * <p>
+                 * See {@link ClientNetworkConfig#connectionAttemptLimit}.
+                 *
+                 * @return connection attempt period in millis
+                 */
+                int32_t getConnectionAttemptPeriod() const;
+
+                /**
+                 * Period for the next attempt to find a member to connect.
+                 *
+                 * @param connectionAttemptPeriod time to wait before another attempt in millis
+                 * @return configured {@link ClientNetworkConfig} for chaining
+                 */
+                ClientNetworkConfig  &setConnectionAttemptPeriod(int32_t connectionAttemptPeriod);
+
+                /**
+                 * Returns the list of candidate addresses that client will use to establish initial connection
+                 *
+                 * @return list of addresses
+                 */
+                std::vector<Address> getAddresses() const;
+
+                /**
+                 * Adds given addresses to candidate address list that client will use to establish initial connection
+                 *
+                 * @param addresses to be added to initial address list
+                 * @return configured {@link com.hazelcast.client.config.ClientNetworkConfig} for chaining
+                 */
+                ClientNetworkConfig &addAddresses(const std::vector<Address> &addresses);
+
+                /**
+                 * Adds given addresses to candidate address list that client will use to establish initial connection
+                 *
+                 * @param addresses to be added to initial address list
+                 * @return configured {@link ClientNetworkConfig} for chaining
+                 */
+                ClientNetworkConfig &setAddresses(const std::vector<Address> &addresses);
+
+                /**
+                * Adds given address to candidate address list that client will use to establish initial connection
+                *
+                * @param address to be added to initial address list
+                * @return configured {@link ClientNetworkConfig} for chaining
+                */
+                ClientNetworkConfig &addAddress(const Address &address);
+
             private:
+                static int32_t CONNECTION_ATTEMPT_PERIOD;
+
                 config::SSLConfig sslConfig;
                 config::ClientAwsConfig clientAwsConfig;
 
                 int64_t connectionTimeout;
+                bool smartRouting;
+
+                int32_t connectionAttemptLimit;
+                int32_t connectionAttemptPeriod;
+
+                std::vector<Address> addressList;
+
+                void updateConnectionTimeout();
```

Comment:
This method seems wrong. connectionTimeout is not related to attemptlimit and attemptPeriod

Reply:
Removed it.